src/main/groovy/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepository.groovy

```diff
@@ -159,38 +159,50 @@ class GitRepository implements ScmRepository {
 
     @Override
     ScmPosition currentPosition(Pattern pattern) {
-        return currentPosition(pattern, Pattern.compile('$a^'))
+        return currentPosition(pattern, Pattern.compile('$a^'), LAST_TAG_SELECTOR)
     }
-    
+
     @Override
-    ScmPosition currentPosition(Pattern pattern, Pattern inversePattern) {
+    ScmPosition currentPosition(Pattern pattern, Closure<String> tagSelector) {
+        return currentPosition(pattern, Pattern.compile('$a^'), tagSelector)
+    }
+
+    @Override
+    ScmPosition currentPosition(Pattern pattern, Pattern inversePattern, Closure<String> tagSelector) {
         if(!hasCommits()) {
             return ScmPosition.defaultPosition()
         }
 
         Map tags = repository.tag.list()
                 .grep({ def tag = it.fullName.substring(GIT_TAG_PREFIX.length()); tag ==~ pattern && !(tag ==~ inversePattern) })
-                .inject([:], { map, entry -> map[entry.commit.id] = entry.fullName.substring(GIT_TAG_PREFIX.length()); return map; })
+                .inject([:], { map, entry ->
+                    if (map[entry.commit.id] == null) {
+                        map[entry.commit.id] = [] as List<String>
+                    }
```

Comment:
I looked for something similar. But that would break the null-check later on, which should check for empty list instead (i.e. just do if (map[id]) return )